packages/core-state/src/wallets/wallet-repository.ts

```diff
@@ -139,15 +139,25 @@ export class WalletRepository implements Contracts.State.WalletRepository {
     }
 
     public forgetByIndex(indexName: string, key: string): void {
+        const forgottenWallet = this.getIndex(indexName).get(key);
+        for (const index of Object.values(this.indexes)) {
+            index.entries().forEach(([name, wallet]) => {
```

Comment:
`forEach` should be avoided in favour of `for (const ... of ...)` due to performance implications on larger datasets.

Reply:
Resolved here => cf1e3ddf8df5f90282c8f174c3cef6885229cc2d